sm64-tas-scripting/Inputs.cpp

```diff
@@ -265,60 +294,56 @@ std::pair<int8_t, int8_t> Inputs::GetClosestInputByYawExact(int16_t intendedYaw,
 	return closestInput;
 }
 
-std::pair<int16_t, float> Inputs::GetIntendedYawMagFromInput(int8_t stickX, int8_t stickY, int16_t cameraYaw)
+std::pair<int16_t, float> Inputs::GetIntendedYawMagFromInput(
+	int8_t stickX, int8_t stickY, int16_t cameraYaw)
 {
 	int16_t baseIntendedYaw = inputsToYawMag.at(stickX).at(stickY).first;
-	float intendedMag = inputsToYawMag.at(stickX).at(stickY).second;
+	float intendedMag				= inputsToYawMag.at(stickX).at(stickY).second;
+
+	return {baseIntendedYaw + cameraYaw, intendedMag};
+}
 
-	return { baseIntendedYaw + cameraYaw, intendedMag };
+bool Inputs::HauEquals(int16_t angle1, int16_t angle2)
+{
+	int16_t hau1 = angle1 - (angle1 & 15);
+	int16_t hau2 = angle2 - (angle2 & 15);
+
+	return hau1 == hau2;
 }
 
 int M64::load()
 {
-	FILE* f;
+	std::ifstream f(fileName.c_str(), ios_base::binary);
+	f.exceptions(ios_base::failbit | ios_base::badbit);
 
 	uint16_t buttons;
 	int8_t stick_x, stick_y;
 
 	size_t err;
 
-	try {
-		if ((err = fopen_s(&f, fileName, "rb")) != 0) {
-			std::cerr << "Bad open of file " << fileName << " Error: " << err << std::endl;
-			exit(EXIT_FAILURE);
-		}
-
-		fseek(f, 0x400, SEEK_SET);
+	try
+	{
+		f.seekg(0x400, ios_base::beg);
 
 		uint64_t index = 0;
-		while (true) {
+		while (true)
+		{
 			uint16_t bigEndianButtons;
+			
```

Comment:
```suggestion
			if (f.peek() == EOF)
				break;
```

If you are up to the end of the file, but have not read it yet this can cause errors